app/UI/nav.tsx

```diff
@@ -80,11 +81,29 @@ export class NavBar extends Component<Props, State> {
         mobileContainer: `h-[4.5rem] flex justify-evenly bg-[${PrimaryBackgroundColor}] items-start pt-2`,
     };
 
+    initialActiveIndex = () => {
+        switch (this.props.activeNav) {
+            case "Social":
+                return 0;
+            case "DM":
+                return 1;
+            case "Profile":
+                return 2;
+            case "About":
+                return 3;
+            case "Setting":
+                return 4;
+            default:
+                return 0;
+        }
+    };
+
     state: State = {
         selectedRelay: "",
-        activeIndex: 0,
+        activeIndex: this.initialActiveIndex(),
```

Comment:
maybe `activeIndex` itself is a bad idea. Always prefer to string over number for indexing.

Reply:
Okay, I replaced it with `activeNav`. All taken from the model(global state).